src/App.tsx

```diff
@@ -1,34 +1,106 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { SelectedTodo } from './types/CurrentTodo';
+import { Select } from './types/Select';
+import { Visibility } from './types/Visibility';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [isVisible, setIsVisible] = useState<Visibility>({
+    visible: false,
+    id: 0,
+  });
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
```

Comment:
You dont use await, so async is redundant
```suggestion
    function fetchData() {
```